fabric-screen-api-v1/src/client/java/net/fabricmc/fabric/impl/client/screen/ScreenEventFactory.java

```diff
@@ -212,4 +222,145 @@ public static Event<ScreenMouseEvents.AfterMouseScroll> createAfterMouseScrollEv
 
 	private ScreenEventFactory() {
 	}
+
+	private static final Set<Screen> ACTIVE_SCREENS = new ReferenceArraySet<>(2);
+
+	public static void activateScreen(Screen screen) {
+		ACTIVE_SCREENS.add(screen);
+	}
+
+	public static void initializeQuiltEventBridges(ModContainer mod) {
+		// Our events are for all screens, while Fabric's are per screen. This ensures that the Quilt events are only run for the right Fabric screen.
```

Comment:
in the future it might be worth adding per-screen events to qsl so they needn't be implemented here